src/rdkafka_lz4.c

```diff
@@ -225,13 +225,15 @@ rd_kafka_lz4_decompress (rd_kafka_broker_t *rkb, int proper_hc, int64_t Offset,
 
         /* Allocate output buffer, we increase this later if needed,
          * but hopefully not. */
-        out = rd_malloc(estimated_uncompressed_size);
-        if (!out) {
+        rd_bool_t succeeded;
+        rd_wait_mem_space(estimated_uncompressed_size, succeeded);
+        if (succeeded == rd_false || !(out = rd_malloc(estimated_uncompressed_size))) {
                 rd_rkb_log(rkb, LOG_WARNING, "LZ4DEC",
                            "Unable to allocate decompression "
                            "buffer of %"PRIusz" bytes: %s",
                            estimated_uncompressed_size, rd_strerror(errno));
                 err = RD_KAFKA_RESP_ERR__CRIT_SYS_RESOURCE;
+                rd_restore_total_mem_usage(estimated_uncompressed_size);
```

Comment:
Why do restore here if we were unsuccessful -  `rk_total_message_mem` should not have changed in this case?

Reply:
I think I made a mistake. We only restore when `succeeded` is `true`